libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -877,7 +877,7 @@ class MapboxNavigation @VisibleForTesting internal constructor(
                                     processedRoute.route.routeId == passedRoute.id
                                 }
                             }
-                        directionsSession.setRoutes(routes, setRoutesInfo)
+                        directionsSession.setRoutes(processedRoutes.routes, setRoutesInfo)
```

Comment:
There is also `setRoutesToTripSession` method. It is only called with NEW reason, but I think we should also use `processedRoutes.routes` there for consistency.

Reply:
I had a code that changed the `setRoutesToTripSession` as well but opted to remove it. This method sets the native object that we have back to Nav Native, so there's no chance that something has changed after it's set.

It's somewhat of an assumption but since rebuilding the road objects is not cheap, I opted to skip this changes since it has no practical value right now. We can consider revisiting it when needed.

Reply:
> since rebuilding the road objects is not cheap

But you won't refresh them, will you? They are only refreshed in `MapboxTripSession#setRoutes` only when the reason is REFRESH. There it's NEW. So `processedRoutes.routes` is absolutely the same list (the same reference even) as `routes`.
This discussion does not block the merge.

Reply:
Ah sorry, you mean `resetTripSessionRoutes`? I thought you mentioned the logic for new route in `MapboxTripSession`.

I guess we could reset the route back to the `MapboxDirectionsSession` there as well, just to keep the `MapboxNavigation` logic consistent and let only `MapboxTripSession` decide what to do functionally.

Reply:
I'll follow up separately.

Reply:
Hmm, this one is more challenging because it resets the routes when session is started. At this point we don't want to trigger the `RoutesObserver`. This might need more investigation on how to handle this situation, I guess it was introduced as a workaround for missing instructions.

Reply:
I only meant that this line
```
                routeAlternativesController.processAlternativesMetadata(
                    routes,
                    nativeAlternatives
                )
```
should use the routes from `NativeSetRouteResult`, not from the method arguments.